src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -16,170 +15,160 @@
+    EVENT_RAISE_DEAD    = 4
+};
 
-class boss_baron_rivendare : public CreatureScript
+uint32 const RaiseDeadSpells[6] =
+{
+    SPELL_RAISE_DEAD_1, SPELL_RAISE_DEAD_2, SPELL_RAISE_DEAD_3,
+    SPELL_RAISE_DEAD_4, SPELL_RAISE_DEAD_5, SPELL_RAISE_DEAD_6
+};
+
+struct boss_baron_rivendare : public BossAI
 {
 public:
-    boss_baron_rivendare() : CreatureScript("boss_baron_rivendare") { }
+    boss_baron_rivendare(Creature* creature) : BossAI(creature, TYPE_BARON), RaiseDead(false) { }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void Reset() override
     {
-        return GetStratholmeAI<boss_baron_rivendareAI>(creature);
+        // needed until re-write of instance scripts is done
+        if (instance->GetData(TYPE_RAMSTEIN) == DONE)
+            instance->SetData(TYPE_BARON, NOT_STARTED);
+
+        BossAI::Reset();
     }
 
-    struct boss_baron_rivendareAI : public ScriptedAI
+    void JustEngagedWith(Unit* who) override
     {
-        boss_baron_rivendareAI(Creature* creature) : ScriptedAI(creature)
-        {
-            Initialize();
-            instance = me->GetInstanceScript();
-        }
+        // needed until re-write of instance scripts is done
+        if (instance->GetData(TYPE_BARON) == NOT_STARTED)
+            instance->SetData(TYPE_BARON, IN_PROGRESS);
 
-        void Initialize()
-        {
-            ShadowBolt_Timer = 5000;
-            Cleave_Timer = 8000;
-            MortalStrike_Timer = 12000;
-            //        RaiseDead_Timer = 30000;
-            SummonSkeletons_Timer = 34000;
-        }
+        events.ScheduleEvent(EVENT_SHADOWBOLT, 5s);
+        events.ScheduleEvent(EVENT_CLEAVE, 8s);
+        events.ScheduleEvent(EVENT_MORTALSTRIKE, 12s);
+        events.ScheduleEvent(EVENT_RAISE_DEAD, 15s);
 
-        InstanceScript* instance;
+        BossAI::JustEngagedWith(who);
+    }
 
-        uint32 ShadowBolt_Timer;
-        uint32 Cleave_Timer;
-        uint32 MortalStrike_Timer;
-        //    uint32 RaiseDead_Timer;
-        uint32 SummonSkeletons_Timer;
+    void JustDied(Unit* killer) override
+    {
+        // needed until re-write of instance scripts is done
+        instance->SetData(TYPE_BARON, DONE);
 
-        void Reset() override
-        {
-            Initialize();
-            if (instance->GetData(TYPE_RAMSTEIN) == DONE)
-                instance->SetData(TYPE_BARON, NOT_STARTED);
-        }
+        BossAI::JustDied(killer);
+    }
 
-        void AttackStart(Unit* who) override
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
         {
-            //can't use entercombat(), boss' dmg aura sets near players in combat, before entering the room's door
-            if (instance->GetData(TYPE_BARON) == NOT_STARTED)
-                instance->SetData(TYPE_BARON, IN_PROGRESS);
-            ScriptedAI::AttackStart(who);
+            if (!me->HasAura(SPELL_UNHOLY_AURA))
+                DoCastSelf(SPELL_UNHOLY_AURA);
+            return;
```

Comment:
Put it in creature_template_addon

Reply:
Like this one for Horgus the Ravager ?
```sql
SELECT * FROM `creature_template_addon` WHERE `auras`= 17467;
```
```sql
entry  path_id   mount  bytes1  bytes2   emote  visibilityDistanceType  auras
-----  -------  ------  ------  ------  ------  ----------------------  -----
10946        0       0       0       0       0                       0  17467
```

Reply:
@Killyana , as far as I remember this was the first thing I've tried. The aura doesn't apply.

Edit: I've checked again and it doesn't apply.

Reply:
It works for me
``` sql
DELETE FROM `creature_addon` WHERE `guid` IN(54241);
UPDATE `creature_template_addon` SET `auras`='17467' WHERE `entry` IN (10440);

Reply:
@Killyana Uh ```DELETE FROM `creature_addon` WHERE `guid` IN(54241);``` is something I never did. Now its working fine. Will update the PR.

Thanks a lot <3